Xamarin.Forms.Platform.Android/CollectionView/CarouselViewRenderer.cs

```diff
@@ -172,15 +173,39 @@ protected override void UpdateAdapter()
 			// So we give it an alternate delegate for creating the views
 
 			var oldItemViewAdapter = ItemsViewAdapter;
+			UnsubscribeCollectionItemsSourceChanged(oldItemViewAdapter);
 
 			ItemsViewAdapter = new ItemsViewAdapter<ItemsView, IItemsViewSource>(ItemsView,
 				(view, context) => new SizedItemContentView(Context, GetItemWidth, GetItemHeight));
 
+
 			SwapAdapter(ItemsViewAdapter, false);
 
+			if (ItemsViewAdapter?.ItemsSource is ObservableItemsSource observableItemsSource)
+				observableItemsSource.CollectionItemsSourceChanged += CollectionItemsSourceChanged;
+
 			oldItemViewAdapter?.Dispose();
 		}
 
+		void UnsubscribeCollectionItemsSourceChanged(ItemsViewAdapter<ItemsView, IItemsViewSource> oldItemViewAdapter)
+		{
+			if (oldItemViewAdapter?.ItemsSource is ObservableItemsSource oldObservableItemsSource)
+				oldObservableItemsSource.CollectionItemsSourceChanged -= CollectionItemsSourceChanged;
+		}
+
+		void CollectionItemsSourceChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+		{
+			var centerItemIndex = -1;
+
+			if (GetLayoutManager() is LinearLayoutManager linearLayoutManager)
+			{
+				var firstVisibleItemIndex = linearLayoutManager.FindFirstVisibleItemPosition();
+				centerItemIndex = RecyclerExtensions.CalculateCenterItemIndex(firstVisibleItemIndex, this, linearLayoutManager);
+			}
+
+			Carousel.SetCurrentItem(null, centerItemIndex);
```

Comment:
Should this be inside the if statement? 

Or instead of using an if cast should it just be an explicit cast because for a CarouselViewRenderer GetLayoutManager()  will always  be a LinearLayoutManager  right?